### PR TITLE
fix: use just one grpc connection for topics

### PR DESCRIPTION
**Pull request by anitarua**
Addresses https://github.com/momentohq/dev-eco-issue-tracker/issues/503

Publish and Subscribe now share grpc connections rather than having one for each.

### CODE REVIEW

File: internal/grpcmanagers/topic_manager.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ func NewStreamTopicGrpcManager(request *models.TopicStreamGrpcManagerRequest) (*
 		endpoint,
 		grpc.WithTransportCredentials(credentials.NewTLS(config)),
 		grpc.WithChainStreamInterceptor(interceptor.AddStreamHeaderInterceptor(authToken)),
+		grpc.WithChainUnaryInterceptor(interceptor.AddHeadersInterceptor(authToken)),
```
Comment (anitarua) [inline internal/grpcmanagers/topic_manager.go:30]: adds authorization header to any unary requests made using this connection

File: momento/pubsub_client.go [hunk 2]
```diff
@@ -44,19 +43,11 @@ func newPubSubClient(request *models.PubSubClientRequest) (*pubSubClient, moment
...
 	}
-
-	unaryDataManager, err := grpcmanagers.NewUnaryDataGrpcManager(&models.DataGrpcManagerRequest{
-		CredentialProvider: request.CredentialProvider,
-		RetryStrategy:      retry.NewNeverRetryStrategy(),
-	})
-	if err != nil {
-		return nil, err
-	}
+	lastStreamTopicManager := streamTopicManagers[numChannels-1]
```
Comment (anitarua) [inline momento/pubsub_client.go:46]: not sure if reserving one specific channel is the best practice, could go the route of the `topicSubscribe` function instead, e.g.
```
topicManager := client.getNextStreamTopicManager()
clientStream, err := topicManager.StreamClient.Publish(...)
```

Comment (pgautier404) [reply momento/pubsub_client.go:46]: I like the approach in your comment better. That `lastStreamTopicManager` is still in the rotation for `getNextStreamTopicManager()` anyhow, so there's really no reason to treat it as a special purpose channel.

**Review by pgautier404 — APPROVED** [review]
LGTM!

**Pull request merged**